### PR TITLE
Type data-bytes of UUIDs.

### PR DESCRIPTION
**Pull request by floitsch**
Fixes https://github.com/toitware/toit/issues/4197.

This is a breaking change as the function used to accept anything that had a `to-byte-array` (like UUIDs themselves), but the documentation always stated "strings or bytearrays".

### CODE REVIEW

File: lib/uuid.toit [hunk 1]
```diff
@@ -97,8 +93,7 @@ UUIDs are equivalent to a 128-bit number. Through the use of
 See https://en.wikipedia.org/wiki/Universally_unique_identifier.
 */
 class Uuid:
-  // TODO(4196): the field should be types as `ByteArray`.
-  bytes_ ::= ?
+  bytes_/io.Data ::= ?
```
Comment (kasperl) [inline lib/uuid.toit:96]: Why is this not a byte array?

Comment (floitsch) [reply lib/uuid.toit]: Changed to ByteArray.
done.

File: lib/uuid.toit [hunk 2]
```diff
@@ -97,8 +93,7 @@ UUIDs are equivalent to a 128-bit number. Through the use of
 See https://en.wikipedia.org/wiki/Universally_unique_identifier.
 */
 class Uuid:
-  // TODO(4196): the field should be types as `ByteArray`.
-  bytes_ ::= ?
+  bytes_/ByteArray ::= ?
```
Comment (kasperl) [inline lib/uuid.toit:96]:
```suggestion
  bytes_/ByteArray
```

File: lib/uuid.toit [hunk 3]
```diff
@@ -107,7 +102,7 @@ class Uuid:
   The given parameter must be a byte array of size 16.
   */
   constructor .bytes_:
-    if bytes_.size != SIZE: throw "INVALID_UUID"
+    if bytes_.byte-size != SIZE: throw "INVALID_UUID"
```
Comment (kasperl) [inline lib/uuid.toit:105]: Why are all these changes to the Uuid class necessary or useful?

File: lib/uuid.toit (hunk 2 above)
Comment (floitsch) [reply lib/uuid.toit:96]: done.

**Review by kasperl — APPROVED** [review]
Looks good to me.

**Pull request merged**